### PR TITLE
fix(resequence): Check the column name from DB

### LINKED ISSUE #1209: pg_query(): Query failed: ERROR:  column author.author_pk does not exist
### Description

I don't know, which version it was, but I had an older version of Fossology repo and therefore docker container, which worked without any problems. Yesterday I updated the Fossology repo and created new docker container. Before, I deleted the old ones. 

**One important note**: I have an external PostgreSQL database. So I changed the `docker-compose.yml`, so all DB relevant variables are pointint to the external PostgreSQL database and removed the link to the Docker Postgre container. This database was also used by the old Fossology version.

Now I wanted to run another scan, but nothing happens, i.e. no progress:

fossology-not-progress

When I call `docker ps`, I get this output:

```
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                         PORTS                  NAMES
5317b1eb6c95        fossology           "/fossology/docker..."   17 hours ago        Up 17 hours                    0.0.0.0:8081->80/tcp   fossology_web_1
bf0fd13709cf        fossology           "/fossology/docker..."   17 hours ago        Restarting (255) 2 hours ago                          fossology_scheduler_1
```

#### How to reproduce

Update from an older Fossology version to the latest. As said before, the old version was linked to an external PostgreSQL database and now it uses the same external PostgreSQL database.

### Versions

* Last commit id on master:
`git rev-parse HEAD` gives me `4580ed337fa1cdda26b11d856550876d29f16203`
* Operating System (lsb_release -a):
On my system, `lsb_release` is not there, but when it helps, here is the output of `cat /etc/*-release`:

> NAME="Red Hat Enterprise Linux Server"
> VERSION="7.5 (Maipo)"
> ID="rhel"
> ID_LIKE="fedora"
> VARIANT="Server"
> VARIANT_ID="server"
> VERSION_ID="7.5"
> PRETTY_NAME="Red Hat Enterprise Linux"
> ANSI_COLOR="0;31"
> CPE_NAME="cpe:/o:redhat:enterprise_linux:7.5:GA:server"
> HOME_URL="https://www.redhat.com/"
> BUG_REPORT_URL="https://bugzilla.redhat.com/"
> 
> REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
> REDHAT_BUGZILLA_PRODUCT_VERSION=7.5
> REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
> REDHAT_SUPPORT_PRODUCT_VERSION="7.5"
> Red Hat Enterprise Linux Server release 7.5 (Maipo)
> Red Hat Enterprise Linux Server release 7.5 (Maipo)

#### Logs

The `docker logs bf` (i.e. the _scheduler_ container) command for the scheduler container gives me this output:

```
NOTE: using external DB on host some.host.local
*** update the database and license_ref table ***
Old release was 3.0.1
*** Cleaning author table ***
PHP Warning:  pg_query(): Query failed: ERROR:  column author.author_pk does not exist
LINE 5: WHERE public.author.author_pk = a.author_pk AND p.pfile_pk I...
              ^ in /usr/local/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
Something went wrong. Try running postinstall again!
[2018-09-19 06:09:56] /usr/local/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  column author.author_pk does not exist LINE 5: WHERE public.author.author_pk = a.author_pk AND p.pfile_pk I...               ^ [] []
PHP Warning:  pg_query(): Query failed: ERROR:  column "option_value" of relation "sysconfig" already exists in /usr/local/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-09-19 06:09:57] /usr/local/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  column "option_value" of relation "sysconfig" already exists [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "sysconfig" ADD COLUMN "option_value" character varying(40) DEFAULT NULL
ERROR:  column "option_value" of relation "sysconfig" already exists' in /usr/local/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:
#0 /usr/local/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "sy...')
#1 /usr/local/share/fossology/lib/php/libschema.php(280): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "sy...')
#2 /usr/local/share/fossology/lib/php/libschema.php(142): fo_libschema->applyTables()
#3 /usr/local/lib/fossology/fossinit.php(169): fo_libschema->applySchema('/usr/local/shar...', false, 'fossology', Array)
#4 {main}
  thrown in /usr/local/share/fossology/lib/php/Db/DbManager.php on line 140

```

#### Fossology logs

Cannot enter in the _scheduler_ container, because this container is stopped.

### LINKED ISSUE #1227: Installing 3.4.0-RC1 packages  on 3.1 crashes at DB update
Attempting to install the packages found here:

https://github.com/fossology/fossology/releases/tag/3.4.0-rc1

onto a 3.1 system crashes with the following installation log:

```
*** Running postinstall for web-only actions***
*** update the database and license_ref table ***
Old release was 3.0.1
*** Cleaning author table ***
PHP Warning:  pg_query(): Query failed: ERROR:  column author.author_pk does not exist
LINE 5: WHERE public.author.author_pk = a.author_pk AND p.pfile_pk I...
              ^ in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-22 01:38:26] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  column author.author_pk does not exist                                                                   LINE 5: WHERE public.author.author_pk = a.author_pk AND p.pfile_pk I...               ^ [] []
Something went wrong. Try running postinstall again!
*** Cleaning tables for new constraints ***
Removed 917 rows from tables with new constraints
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "clearing_decision_event" violates foreign key                                                                   constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event". in /usr/share/fossology/lib/php/Db/Driver/                                                                  Postgres.php on line 81
[2018-10-22 01:38:27] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "clearing_de                                                                  cision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey" DETAIL:  Key (clearing_event_fk)=(2                                                                  ) is not present in table "clearing_event". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "clearing_decis                                                                  ion_event" ADD CONSTRAINT "clearing_decision_event_event_fk_fkey" FOREIGN KEY ("clearing_event_fk") REFERENCES "clearing_                                                                  event" ("clearing_event_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_even                                                                  t_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event".' in /usr/share/fossology/lib/php/Db/DbMana                                                                  ger.php:140
Stack trace:
#0 /usr/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "cl...                                                                  ')
#1 /usr/share/fossology/lib/php/libschema.php(70): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "cl...', 'fo_libsch                                                                  ema::a...')
#2 /usr/share/fossology/lib/php/libschema.php(465): fo_libschema->applyOrEchoOnce('ALTER TABLE "cl...', 'fo_libschema::a.                                                                  ..')
#3 /usr/share/fossology/lib/php/libschema.php(152): fo_libschema->appl in /usr/share/fossology/lib/php/Db/DbManager.php o                                                                  n line 140
dpkg: error processing package fossology-web (--install):
 subprocess installed post-installation script returned error exit status 255
Setting up fossology-wgetagent (3.4.0~rc1-1) ...
dpkg: dependency problems prevent configuration of fossology:
 fossology depends on fossology-web; however:
  Package fossology-web is not configured yet.
 fossology depends on fossology-scheduler; however:
  Package fossology-scheduler is not configured yet.

dpkg: error processing package fossology (--install):
 dependency problems - leaving unconfigured
Setting up fossology-buckets (3.4.0~rc1-1) ...
Processing triggers for man-db (2.7.0.2-5) ...
Processing triggers for systemd (215-17+deb8u7) ...
Errors were encountered while processing:
 fossology-ninka
 fossology-scheduler
 fossology-web
 fossology

```

### PR DESCRIPTION
**Pull request by GMishx**
## Description

After renaming the primary column names of `copyright`, `author`, `ecc` and `keyword` table, the migration script was failing.

### Changes

Check the copyright and author table column name from DB before resequencing the tables.
Also check for tables before applying new constraints (which does not exists in previous versions).

## How to test

Install fossology and run post install script on existing old setup.

```bash
$ sudo make install
$ sudo install/fo-postinstall
```

Closes #1227
Closes #1209

### CODE REVIEW

File: install/fossinit.php [hunk 1]
```diff
@@ -163,6 +163,7 @@ function explainUsage()
 
 if($dbManager->existsTable("author"))
 {
+  $GLOBALS["DBName"] = $DatabaseName;
```
[the remark this answers is not in the document]
Comment (GMishx) [reply install/fossinit.php]: Moved

**Comment by mcjaeger** [comment]
I think I am going to test it with updating an existing instance now

**Comment by shaheemazmalmmd** [comment]
Please fix the below php notices 
screenshot

To reproduce:

Clone the release 3.1 the update with the latest code.

**Comment by GMishx** [comment]
Silly me, blindly copy pasting code 😛

**Comment by mcjaeger** [comment]
I did the following.

* I build the debian packages from thebranch
* I copied the debian packaged to a fossology wehere I have installed the branch from his PR before. https://github.com/fossology/fossology/pull/1228
* I installed these packages with the following result:

````
PHP Notice:  Undefined variable: tableToClean in /usr/lib/fossology/dbmigrate_3.3-3.4.php on line 89
Removed 0 rows from tables with new constraints
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event". in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-24 04:58:59] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey" DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "clearing_decision_event" ADD CONSTRAINT "clearing_decision_event_event_fk_fkey" FOREIGN KEY ("clearing_event_fk") REFERENCES "clearing_event" ("clearing_event_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event".' in /usr/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:
#0 /usr/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "cl...')
#1 /usr/share/fossology/lib/php/libschema.php(70): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#2 /usr/share/fossology/lib/php/libschema.php(465): fo_libschema->applyOrEchoOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#3 /usr/share/fossology/lib/php/libschema.php(152): fo_libschema->appl in /usr/share/fossology/lib/php/Db/DbManager.php on line 140
dpkg: error processing package fossology-web (--install):
 subprocess installed post-installation script returned error exit status 255
Setting up fossology-wgetagent (3.2.0-1) ...
dpkg: dependency problems prevent configuration of fossology:

````

Question: is this branch actually rebased now?

**Comment by mcjaeger** [comment]
I think the error still exists with an updated branch like at `67dcbce`

```
*** update the database and license_ref table ***
Old release was 3.1.0
*** Cleaning tables for new constraints ***
Removed 0 rows from tables with new constraints
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event". in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-24 05:16:30] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey" DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "clearing_decision_event" ADD CONSTRAINT "clearing_decision_event_event_fk_fkey" FOREIGN KEY ("clearing_event_fk") REFERENCES "clearing_event" ("clearing_event_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(2) is not present in table "clearing_event".' in /usr/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:
#0 /usr/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "cl...')
#1 /usr/share/fossology/lib/php/libschema.php(70): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#2 /usr/share/fossology/lib/php/libschema.php(465): fo_libschema->applyOrEchoOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#3 /usr/share/fossology/lib/php/libschema.php(152): fo_libschema->appl in /usr/share/fossology/lib/php/Db/DbManager.php on line 140
dpkg: error processing package fossology-web (--install):
 subprocess installed post-installation script returned error exit status 255

```

**Comment by mcjaeger** [comment]
testing on another instance at `Version: [3.2.0rc1], Branch: [master], Commit: [#191f35] `
not working as well:

```
*** Setting up scheduler ***
FOSSology postinstall complete, but sure to complete the remaining
  steps in the INSTALL instructions.
Setting up fossology-web (3.2.0-1) ...
*** Running postinstall for web-only actions***
*** update the database and license_ref table ***
Old release was 3.0.1
*** Cleaning author table ***
*** Cleaning copyright table ***
*** Resequencing author table ***
*** Resequencing copyright table ***
PHP Warning:  pg_query(): Query failed: ERROR:  relation "public.copyright_pk_seq" does not exist in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-24 16:00:57] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  relation "public.copyright_pk_seq" does not exist [] []
Something went wrong. Try running postinstall again!
*** Cleaning tables for new constraints ***
Removed 0 rows from tables with new constraints
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_decision_fk_fkey"
DETAIL:  Key (clearing_decision_fk)=(2) is not present in table "clearing_decision". in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-24 16:00:59] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_decision_fk_fkey" DETAIL:  Key (clearing_decision_fk)=(2) is not present in table "clearing_decision". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "clearing_decision_event" ADD CONSTRAINT "clearing_decision_event_decision_fk_fkey" FOREIGN KEY ("clearing_decision_fk") REFERENCES "clearing_decision" ("clearing_decision_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_decision_fk_fkey"
DETAIL:  Key (clearing_decision_fk)=(2) is not present in table "clearing_decision".' in /usr/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:
#0 /usr/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "cl...')
#1 /usr/share/fossology/lib/php/libschema.php(70): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#2 /usr/share/fossology/lib/php/libschema.php(465): fo_libschema->applyOrEchoOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#3 /usr/share/fossology/lib/php/libschema.php(152 in /usr/share/fossology/lib/php/Db/DbManager.php on line 140
dpkg: error processing package fossology-web (--configure):
 subprocess installed post-installation script returned error exit status 255

```

**Comment by mcjaeger** [comment]
I think the branch is not mergeale because updates on existing instances fail

**Comment by mcjaeger** [comment]
testing with one older instance yields this (not working), new constraint failing:
```
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "ecc" violates foreign key constraint "ecc_pfile_fk_fkey"
DETAIL:  Key (pfile_fk)=(106) is not present in table "pfile". in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-29 20:16:12] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "ecc" violates foreign key constraint "ecc_pfile_fk_fkey" DETAIL:  Key (pfile_fk)=(106) is not present in table "pfile". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "ecc" ADD CONSTRAINT "ecc_pfile_fk_fkey" FOREIGN KEY ("pfile_fk") REFERENCES "pfile" ("pfile_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "ecc" violates foreign key constraint "ecc_pfile_fk_fkey"
DETAIL:  Key (pfile_fk)=(106) is not present in table "pfile".' in /usr/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:

```

**Comment by mcjaeger** [comment]
tested also with `Version: [3.2.0rc1], Branch: [master], Commit: [#191f35] 2017/10/24 05:51 UTC built @ 2017/10/25 09:23 UTC`:
````
Setting up fossology-web (3.2.0-1) ...
*** Running postinstall for web-only actions***
*** update the database and license_ref table ***
Old release was 3.0.1
*** Cleaning author table ***
*** Cleaning copyright table ***
*** Resequencing author table ***
*** Resequencing copyright table ***
*** Cleaning tables for new constraints ***
Removed 0 rows from tables with new constraints
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "ecc" violates foreign key constraint "ecc_pfile_fk_fkey"
DETAIL:  Key (pfile_fk)=(106) is not present in table "pfile". in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-29 21:03:28] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "ecc" violates foreign key constraint "ecc_pfile_fk_fkey" DETAIL:  Key (pfile_fk)=(106) is not present in table "pfile". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "ecc" ADD CONSTRAINT "ecc_pfile_fk_fkey" FOREIGN KEY ("pfile_fk") REFERENCES "pfile" ("pfile_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "ecc" violates foreign key constraint "ecc_pfile_fk_fkey"
DETAIL:  Key (pfile_fk)=(106) is not present in table "pfile".' in /usr/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:
#0 /usr/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "ec...')
#1 /usr/share/fossology/lib/php/libschema.php(70): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "ec...', 'fo_libschema::a...')
#2 /usr/share/fossology/lib/php/libschema.php(465): fo_libschema->applyOrEchoOnce('ALTER TABLE "ec...', 'fo_libschema::a...')
#3 /usr/share/fossology/lib/php/libschema.php(152): fo_libschema->applyConstraints()
#4 /usr/lib/fossology/fossinit.php(175): fo_libschema->applySchema('/usr/share/foss...', false, 'fossology', in /usr/share/fossology/lib/php/Db/DbManager.php on line 140
dpkg: error processing package fossology-web (--configure):
 subprocess installed post-installation script returned error exit status 255
Setting up libboost-program-options1.55.0:amd64 (1.55.0-1) ...
Setting up libboost-regex1.55.0:amd64 (1.55.0-1) ...

````

**Comment by mcjaeger** [comment]
test again

````
*** update the database and license_ref table ***
Old release was 3.0.1
*** Cleaning author table ***
*** Cleaning copyright table ***
*** Resequencing author table ***
*** Resequencing copyright table ***
*** Cleaning tables for new constraints ***
Table keyword or agent does not exists, not cleaning!
Table keyword or pfile does not exists, not cleaning!
Table keyword_decision or pfile does not exists, not cleaning!
Removed 3255 rows from tables with new constraints
PHP Warning:  pg_query(): Query failed: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(183) is not present in table "clearing_event". in /usr/share/fossology/lib/php/Db/Driver/Postgres.php on line 81
[2018-10-30 21:57:43] /usr/share/fossology/lib/php/libschema.php.CRITICAL: ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey" DETAIL:  Key (clearing_event_fk)=(183) is not present in table "clearing_event". [] []
PHP Fatal error:  Uncaught exception 'Fossology\Lib\Exception' with message 'error executing: ALTER TABLE "clearing_decision_event" ADD CONSTRAINT "clearing_decision_event_event_fk_fkey" FOREIGN KEY ("clearing_event_fk") REFERENCES "clearing_event" ("clearing_event_pk") ON UPDATE NO ACTION ON DELETE CASCADE;

ERROR:  insert or update on table "clearing_decision_event" violates foreign key constraint "clearing_decision_event_event_fk_fkey"
DETAIL:  Key (clearing_event_fk)=(183) is not present in table "clearing_event".' in /usr/share/fossology/lib/php/Db/DbManager.php:140
Stack trace:
#0 /usr/share/fossology/lib/php/Db/DbManager.php(204): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "cl...')
#1 /usr/share/fossology/lib/php/libschema.php(70): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#2 /usr/share/fossology/lib/php/libschema.php(465): fo_libschema->applyOrEchoOnce('ALTER TABLE "cl...', 'fo_libschema::a...')
#3 /usr/share/fossology/lib/php/libschema.php(152): fo_libschema->ap in /usr/share/fossology/lib/php/Db/DbManager.php on line 140
dpkg: error processing package fossology-web (--configure):
 subprocess installed post-installation script returned error exit status 255
````

**Comment by mcjaeger** [comment]
also when opening fossology clearing decision with custom license text and comment is lost after opening fossology again. so data is lost.

**Pull request merged**